service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/DocmosisGeneratedOrder.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.fpl.model.docmosis;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+import uk.gov.hmcts.reform.fpl.enums.EPOType;
+import uk.gov.hmcts.reform.fpl.enums.GeneratedOrderType;
+
+import java.util.List;
+
+@Getter
+@SuperBuilder
+@EqualsAndHashCode(callSuper = true)
+public class DocmosisGeneratedOrder extends DocmosisOrder {
```

Comment:
so we need to do some work around docmosis order. It is inheriting a lot of fields it does not use and I do not think this is correct. My plan was to extract out what is truely common between orders as I refactored these

Reply:
Ok I will check and ensure only the common fields are retained in DocmosisOrder.

Reply:
Done

Reply:
Maybe the `DocmosisData` class can be changed from an interface to an abstract class and it can contain the variables for all the relevant images (draft background, court seal, etc)?

Reply:
I wonder if this can be handled in a separate ticket. As DocmosisCoverDocument currently implements DocmosisData but does not use the image variables. Hence not all classes which implement DocmosisData use image variables. The proper class hierarchy needs to be revisited.

Reply:
@KiranVarma02 made this ticket https://tools.hmcts.net/jira/browse/FPLA-1793

Reply:
I'm fine to look at this retrospectively, but yeah I think that 1793 is SUPER important now. We want this to be nice strongly typed objects with the correct fields otherwise it's not as beneficial as it should be